app/routes/datamart/land.py

```diff
@@ -41,28 +45,107 @@
+        )
+
+    # Merge dataset version overrides with default dataset versions
+    return DEFAULT_LAND_DATASET_VERSIONS | dataset_versions
+
+
+def _parse_area_of_interest(request: Request) -> AreaOfInterest:
+    params = request.query_params
+    aoi_type = params.get('aoi[type]')
+    try:
+        if aoi_type == 'geostore':
+            return GeostoreAreaOfInterest(geostore_id=params.get('aoi[geostore_id]', None))
+
+            # Otherwise, check if the request contains admin area information
+        if aoi_type == 'admin':
+            return AdminAreaOfInterest(
+                country=params.get('aoi[country]', None),
+                region=params.get('aoi[region]', None),
+                subregion=params.get('aoi[subregion]', None),
+                provider=params.get('aoi[provider]', None),
+                version=params.get('aoi[version]', None),
+            )
+
+        # If neither type is provided, raise an error
+        raise HTTPException(status_code=422, detail="Invalid Area of Interest parameters")
+    except ValidationError as e:
+        raise HTTPException(status_code=422, detail=e.errors())
+
 
 @router.get(
     "/tree_cover_loss_by_driver",
     response_class=ORJSONResponse,
     response_model=DataMartResourceLinkResponse,
     tags=["Land"],
     status_code=200,
+    openapi_extra={
+        "parameters": [
+            {
+                "name": "aoi",
+                "in": "query",
+                "required": True,
+                "style": "deepObject",
+                "explode": True,
+                "example": {
+                    "geostore_id": "637d378f-93a9-4364-bfa8-95b6afd28c3a",
+                },
+                "description": "The Area of Interest",
+                "schema": {
+                    "oneOf": [
+                        {"$ref": "#/components/schemas/GeostoreAreaOfInterest"},
+                        {"$ref": "#/components/schemas/AdminAreaOfInterest"},
+                    ]
+                }
+            },
+            {
+                "name": "dataset_version",
+                "in": "query",
+                "required": False,
+                "style": "deepObject",
+                "explode": True,
+                "schema": {
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                },
+                "example": {
+                    "umd_tree_cover_loss": "v1.11",
+                    "tsc_tree_cover_loss_drivers": "v2023",
+                },
+                "description": (
+                        "Pass dataset version overrides as bracketed query parameters.",
+                )
+            }
+        ]
+    },
 )
 async def tree_cover_loss_by_driver_search(
     *,
-    request: Request,
-    geostore_id: UUID = Query(..., title="Geostore ID"),
+    aoi: AreaOfInterest = Depends(_parse_area_of_interest),
```

Comment:
I feel like `aoi` is a very acronym internally, but I'm not sure about externally. Should we make it more explicit, e.g. `area_of_interest`? Or do you think it's fine as long as it's clearly used as terminology in the docs?

Reply:
I think we gain more readability by keeping it short. So, I'd vote for a definition in the docs.

That said, I hear my kids and their friends saying `AOI` or `POI` all the time when gaming. 😉